sunkit_image/tests/test_granule.py

```diff
@@ -0,0 +1,211 @@
+        assert data_map.data[x[i], y[i]] != segmented.data[x[i], y[i]]
+
+    # ------ error raising tests ------ :
+    # Test 5: check that errors are raised for incorrect inputs
+    with pytest.raises(TypeError, match="Input must be sunpy map."):
+        granule.segment(np.array([[1, 2, 3], [1, 2, 3]]), test_method, test_res)
+    with pytest.raises(TypeError, match="Method must be one of: "):
+        granule.segment(data_map, "banana", test_res)
+
+
+def test_get_threshold(inputs):
+    """
+    Unit tests for get_threshold() function.
+    """
+
+    _, _, test_method = inputs
+
+    # -------- positive tests -------- :
+    # Test 1: check type of output
+    test_arr1 = np.array([[1, 2, 3, 4, 5], [1, 2, 3, 4, 5]])
+    threshold1 = granule.get_threshold(test_arr1, test_method)
+    assert type(threshold1) is np.float64
+    # Test 2: check range of output
+    assert 0 < threshold1 < np.max(test_arr1)
+
+    # -------- negative tests -------- :
+    # Test 3: check that different arrays return different thresholds
+    test_arr2 = np.array([[2, 3, 4, 5, 6], [2, 3, 4, 5, 6]])
+    threshold2 = granule.get_threshold(test_arr2, test_method)
+    assert threshold1 != threshold2
+
+    # ------ error raising tests ------ :
+    # Test 4: check that errors are raised for incorrect inputs
+    with pytest.raises(ValueError, match="Input data must be an array."):
+        granule.get_threshold([], test_method)
+    with pytest.raises(ValueError, match="Method must be one of: "):
+        granule.get_threshold(test_arr1, "banana")
+
+
+def test_trim_intergranules(inputs):
+    """
+    Unit tests for trim_intergranules() function.
+    """
+
+    data_map, _, _ = inputs
+
+    # -------- positive tests -------- :
+    thresholded = np.uint8(data_map.data > np.nanmedian(data_map.data))
+    # Test 1: check that returned array is not empty
+    assert np.size(thresholded) > 0
+    # Test 2: check that the correct dimensions are returned
+    assert thresholded.shape == granule.trim_intergranules(thresholded).shape
+    # Test 3: mark erronous material, not remove when flag is True.
+    middles_marked = granule.trim_intergranules(thresholded, mark=True)
+    marked_erroneous = np.count_nonzero(middles_marked[middles_marked == 0.5])
+    assert marked_erroneous != 0
+    # Test 4: remove when flag is False (no 0.5 values)
+    middles_marked = granule.trim_intergranules(thresholded, mark=False)
+    marked_erroneous = np.count_nonzero(middles_marked[middles_marked == 0.5])
+    assert marked_erroneous == 0
+
+    # -------- negative tests -------- :
+    # Test 4: check that the returned array has fewer (or same number)
+    # 0-valued pixels as input array (for a data set which we
+    # know by eye should have some middle sections removed)
+    middles_removed = granule.trim_intergranules(thresholded)
+    assert (np.count_nonzero(middles_removed) < np.count_nonzero(thresholded)) is False
+
+    # ------ error raising tests ------ :

```

Comment:
These should be a separate unit test.

Reply:
Ah I see... everytime the function is called again that should be an entirely seperate test function, not just a seperate test within a test function? I will go through and change these.

Reply:
It depends on the context.

For example the error handling, does not depend on the others parts of the test.

But tests that require some build up, should be left combined.